rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -133,28 +120,129 @@
+      // Get the indexes for selected runs. This should either be [0] or [0, 1].
+      var chartKeys = _.keys(state.selectedRuns);
+
+      // D3 data join for charts. Create a container div for each chart to store graph, select element and buttons.
+      var containers = d3.select(element).select(
+        rapidBlockResultsSel
+      ).selectAll(".chart-container").data(chartKeys);
+      var containersEnter = containers.enter()
+        .append("div");
+
+      // chart selection, close and compare buttons
+      var selectionRows = containers.selectAll(".selection-row");
+      var selectionRowsEnter = containersEnter
+        .append("div")
+        .classed("selection-row", true);
+
+      var select = selectionRowsEnter.append("select")
+        .on('change', function(index) {
+          var selectedRun = select.property('value');
+          if (selectedRun !== NONE_SELECTION) {
+            selectedRun = parseInt(selectedRun);
+          }
+          state.selectedRuns[index] = selectedRun;
+          render();
+        });
+
+      selectionRowsEnter.append("a")
+        .classed("compare-responses", true).text("Compare responses").on("click", function() {
+          state.selectedRuns = [state.selectedRuns[0], NONE_SELECTION];
+          render();
+        });
+
+      selectionRowsEnter.append("a")
+        .classed("close", true).text("Close ").on('click', function(index) {
+          state.selectedRuns.splice(index, 1);
+          render();
+        }).append("span").attr("class", "fa fa-close");
+
+      var selectionRowsMerged = selectionRowsEnter.merge(selectionRows);
+      selectionRowsMerged.selectAll(".compare-responses").classed("hidden", function() {
+        return chartKeys.length !== 1 || state.is_open || state.runs.length < 2;
+      });
+      selectionRowsMerged.selectAll(".close").classed("hidden", function() {
+        return chartKeys.length === 1;
+      });
+
+      // create the chart svg container
+      var chartsEnter = containersEnter
+        .append("svg")
+        // The g element has a little bit of padding so the x and y axes can surround it
+        .append("g").attr("class", "chart");
+      // create x and y axes
+      chartsEnter.append("g").attr("class", "xaxis").attr("transform", "translate(0," + ChartSettings.height + ")");
+      chartsEnter.append("g").attr("class", "yaxis");
+
+      containersEnter.merge(containers)
+        .attr("class", "chart-container " + (chartKeys.length === 1 ? 'single-chart' : 'two-charts'))
+        .each(function (index, __, charts) {
+          renderChart(d3.select(charts[index]), index);
+        })
+        .selectAll("svg")
+        .attr("width", (ChartSettings.width / chartKeys.length) + ChartSettings.left + ChartSettings.right)
+        .attr("height", ChartSettings.height + ChartSettings.top + ChartSettings.bottom)
+        .selectAll(".chart")
+        .attr("transform", "translate(" + ChartSettings.left + "," + ChartSettings.top + ")");
```

Comment:
Wondering if it makes more sense to render the `svg` and the `g` elements within `renderChart` function. I would consider those elements to be part of the chart

Reply:
This is a little more difficult than I anticipated since the chart selector needs to be passed in to work with D3's data join... do you mind if I just leave it as is?

Reply:
Yup, sounds good to me